In [1]:
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import StratifiedShuffleSplit
from sklearn.metrics import *
from sklearn.preprocessing import OrdinalEncoder
import openpyxl
import pandas as pd

In [2]:
df = pd.read_excel(r'E:\datasets\titanic.xlsx')

In [3]:
df.head()

status    age   sex survived
0  first  adult  male      yes
1  first  adult  male      yes
2  first  adult  male      yes
3  first  adult  male      yes
4  first  adult  male      yes

In [4]:
split = StratifiedShuffleSplit(n_splits=1, test_size=0.2, random_state=42)
for train_index, test_index in split.split(df, df["sex"]):
    strat_train_set = df.loc[train_index]
    strat_test_set = df.loc[test_index]

In [5]:
encoder = OrdinalEncoder()
df_train = encoder.fit_transform(strat_train_set)
df_test = encoder.fit_transform(strat_test_set)

In [6]:
clf = RandomForestClassifier().fit(df_train[:, [0, 1, 2]], df_train[:, 3])

In [7]:
test_pred = clf.predict(df_test[:, [0, 1, 2]])
accuracy_score(test_pred.round(), df_test[:, 3]) * 100

76.64399092970523

Accuracy scores:

1. RandomForestRegressor: 76.64399092970523 %
2. LinearRegression: 76.64399092970523 %
3. DecicionTreeClassifier: 76.64399092970523 %
4. SVM: 76.64399092970523 %
5. RandomForestClassifier: 76.64399092970523 %
6. KNeighborsClassifier: 62.585034013605444 %